error.go

```diff
@@ -65,6 +72,18 @@ func (e *Error) UnmarshalJSON(data []byte) error {
 	return nil
 }
 
+// Errors provides a type for a collection of Salesforce Errors
+type Errors []Error
+
+func (e Errors) Error() string {
+	msgs := make([]string, len(e))
+	for i, err := range e {
+		msgs[i] = err.Message
+	}
```

Comment:
Let's use `append` it looks safer.
```suggestion
	msgs := make([]string, 0, len(e))
	for _, err := range e {
		msgs = append(msgs, err.Message)
	}
```